staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -151,7 +149,7 @@ func (t *Transport) rewriteURL(targetURL string, sourceURL *url.URL, sourceReque
 	isDifferentHost := url.Host != "" && url.Host != sourceURL.Host && url.Host != sourceRequestHost
 	isRelative := !strings.HasPrefix(url.Path, "/")
 	if isDifferentHost || isRelative {
-		return targetURL
+		return url.String()
```

Comment:
remind me what the parse / String() round-trip is accomplishing?

Reply:
The round-trip allows returning an escaped version of the URL. For instance, to obtain the path, String uses u.EscapedPath().

Reply:
and is it otherwise identical to the string we used to parse `targetURL` or is there other normalization (like adding host prefixes, etc)

Reply:
yes, the string we used to parse `targetURL` is the base that is now used to build the url.URL object. Besides having some characters escaped, the URL will not change compared to what we had before.